configs/components/puppet.rb

```diff
@@ -59,6 +63,83 @@
     pkg.install_service "ext/solaris/smf/puppet.xml", "ext/solaris/smf/puppet", service_type: "network"
   when "aix"
     pkg.install_service "resources/aix/puppet.service", nil, "puppet"
+  when "windows"
+    description = "Periodically fetches and applies configurations from a Puppet master server."
+    service_hash = [
```

Comment:
I share @Iristyle concerns about providing an abstraction over WiX for a few reasons. My main concern is it will force us to implement everything in order to ship anything.

I'd be a big +1 on passing WiX files to vanagon, rather than vanagon knowing how to generate them (using ERB templates or otherwise). That way we can get MSIs building for projects we already have, validated in CI, and shipped. There will be some copy and paste for new projects, but it's fairly boiler plate code.

Then over time, continue working on splitting functionality into component repos, e.g. facter's WiX component should know to create `facts.d`, and refactoring common functionality where it makes sense.

I realize I'm late to the party and the ship may have already sailed, but I think it would be much less complicated to do this in stages as opposed to trying to create an abstraction over WiX, which is itself an abstraction over Windows Installer. Unfortunately both are difficult to work with, even for simple things.